shared/src/Datadog.Trace.ClrProfiler.Native/process_helper.cpp

```diff
@@ -87,187 +90,60 @@ bool ProcessHelper::RunProcess(const std::string& processPath,
-        // "The implementation interposes the user-installed signal(3) handlers....to postpone signal delivery to threads
-        // which entered (libthr-internal) critical sections..."  We want to pass SIG_DFL anyway.
-        sigset_t junk_signal_set;
-        struct sigaction sa_default;
-        struct sigaction sa_old;
-        memset(&sa_default, 0, sizeof(sa_default)); // On some architectures, sa_mask is a struct so assigning zero to it doesn't compile
-        sa_default.sa_handler = SIG_DFL;
-        for (int sig = 1; sig < NSIG; ++sig)
-        {
-
-            if (sig == SIGKILL || sig == SIGSTOP)
-            {
-                continue;
-            }
-            if (!sigaction(sig, NULL, &sa_old))
-            {
-                void (*oldhandler)(int) = handler_from_sigaction (&sa_old);
-
-                if (oldhandler != SIG_IGN && oldhandler != SIG_DFL)
-                {
-                    // It has a custom handler, put the default handler back.
-                    // We check first to preserve flags on default handlers.
-                    sigaction(sig, &sa_default, NULL);
-                }
-            }
-        }
-
-        pthread_sigmask(SIG_SETMASK, &old_signal_set, &junk_signal_set); // Not all architectures allow NULL here
-
-        // Finally, execute the new process.  execve will not return if it's successful.
-        execve(processPath.c_str(), const_cast<char* const *>(argv.data()), environ);
-        ExitChild(waitForChildToExecPipe[WRITE_END_OF_PIPE], errno); // execve failed
+        Log::Warn("SingleStepGuardRails::SendTelemetry: Failed to initialize the pipe");
+        return false;
     }
 
-    // Restore signal mask in the parent process immediately after fork() or vfork() call
+    int read_fd = pipefd[0];
+    int write_fd = pipefd[1];
 
-    pthread_sigmask(SIG_SETMASK, &old_signal_set, &signal_set);
+    pid_t processId = fork();
 
-    if (processId < 0)
+    if (processId == 0)
     {
+        // Child process
 
-        // failed
-        success = false;
-    }
-
-    // Also close the write end of the exec waiting pipe, and wait for the pipe to be closed
-    // by trying to read from it (the read will wake up when the pipe is closed and broken).
-    // Ignore any errors... this is a best-effort attempt.
-
-    CloseIfOpen(waitForChildToExecPipe[WRITE_END_OF_PIPE]);
-    if (waitForChildToExecPipe[READ_END_OF_PIPE] != -1)
-    {
+        close(write_fd); // The child process doesn't need to write
 
-        int childError;
-        if (success)
+        // Redirect stdin to the read end of the pipe
+        if (dup2(read_fd, STDIN_FILENO) == -1)
         {
-
-            ssize_t result = ReadSize(waitForChildToExecPipe[READ_END_OF_PIPE], &childError, sizeof(childError));
-            if (result == sizeof(childError))
-            {
-                success = false;
-            }
+            _exit(EXIT_FAILURE);
```

Comment:
Just so I understand, we're exiting in a new process now, so this only exits the _child_ right?

Reply:
Correct